### DISCUSSION TITLE
ozone (docker) volume create: "WARN Unable to load native-hadoop library for your platform"

### DISCUSSION THREAD

**Discussion by pjotrek-b** [apache/ozone#6723] [General]
Running `ozone` commands inside a docker image, freshly pulled from DockerHub (according to docs).

When trying to create a volume, I get an error (actually an endless loop, retrying):

`$ ozone sh volume create /vol1`

> 2024-05-23 21:41:38 WARN  NativeCodeLoader:60 - Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
> 2024-05-23 21:41:41 INFO  RetryInvocationHandler:422 - com.google.protobuf.ServiceException: java.net.ConnectException: Call From 20c05fb69b5f/172.17.0.2 to localhost:9862 failed on connection exception: java.net.ConnectException: Connection refused; For more details see:  http://wiki.apache.org/hadoop/ConnectionRefused, while invoking $Proxy21.submitRequest over nodeId=null,nodeAddress=localhost:9862 after 1 failover attempts. Trying to failover after sleeping for 4000ms. Current retry count: 1.

Any help or advice appreciated :smile:

**Comment by pjotrek-b** [comment]
btw: I've read the [Connection Refused](https://cwiki.apache.org/confluence/display/HADOOP2/ConnectionRefused) article linked in the error message.

And indeed, running `telnet localhost 9862` gives me a **connection refused**.

But why?
All I did was run/start the docker container with default settings.

**Comment by adoroszlai** [comment]
> All I did was run/start the docker container with default settings.

Can you please provide the command used?

**Reply by adoroszlai** [reply]
Based on #6721 I guess this is the command:

> `docker run -d -p 9878:9878 -p 9876:9876 apache/ozone`

The problem is that Ozone Manager exits due to misconfiguration.  Filed HDDS-10909 for this bug.  Thank you for reporting the problem.

In the meantime, please try using `docker-compose` (see "Running a pseudo-cluster with multiple containers" section in the same doc).

**Reply by pjotrek-b** [reply]
Yes, that command.
Thanks for drawing the connection between my 2 postings.

Sorry!
I forgot to link [the tutorial documentation](https://cwiki.apache.org/confluence/display/OZONE/Running+via+DockerHub) I was referring to, here too :blush:

**Reply by pjotrek-b** [reply]
I also already used `docker-compose` (pseudo-cluster with 3 nodes, as in the docs) - and it started, but it was too late yesterday for me to try out connections. Will do that.
